holoviews/__init__.py

```diff
@@ -25,6 +25,7 @@
 from .operation import ElementOperation, MapOperation, TreeOperation # noqa (API import)
 from .element import *                                               # noqa (API import)
 from . import util # noqa (API import)
+from .element import __all__ as elements_list
```

Comment:
Appears you were right being hesitant, but it really shouldn't be relying on importing Overlay via ``hv.element``.